File: Easy/attivform/meta_attivformdefaultview/meta_attivformdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_attivformdefaultview
{
    public class Meta_attivformdefaultview : Meta_easydata 
	{
        public Meta_attivformdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "attivformdefaultview") {
				Name = "Attività formative";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"aa","idattivform","idcorsostudio","iddidprog","iddidproganno","iddidprogcurr","iddidprogori","iddidprogporzanno"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "didprogcurr_title", "Curriculum", nPos++);
						DescribeAColumn(T, "didprogori_title", "Orientamento", nPos++);
						DescribeAColumn(T, "didproganno_anno", "Anno di corso", nPos++);
						DescribeAColumn(T, "didprogporzanno_indice", "Indice Porzione d'anno", nPos++);
						DescribeAColumn(T, "didprogporzanno_iddidprogporzannokind", "Durata Porzione d'anno", nPos++);
						DescribeAColumn(T, "didproggrupp_title", "Gruppo opzionale", nPos++);
						DescribeAColumn(T, "insegn_denominazione", "Denominazione Insegnamento", nPos++);
						DescribeAColumn(T, "insegn_codice", "Codice Insegnamento", nPos++);
						DescribeAColumn(T, "insegninteg_denominazione", "Denominazione Integrando", nPos++);
						DescribeAColumn(T, "insegninteg_codice", "Codice Integrando", nPos++);
						DescribeAColumn(T, "attivform_start", "Dal", nPos++);
						DescribeAColumn(T, "attivform_stop", "Al", nPos++);
						DescribeAColumn(T, "attivform_tipovalutaz", "Profitto o Idoneità", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "attivform_sortcode";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/attivform/meta_attivformdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_attivformdefaultview() {
        MetaData.apply(this, ["attivformdefaultview"]);
        this.name = 'meta_attivformdefaultview';
    }

    meta_attivformdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_attivformdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'didprogcurr_title', 'Curriculum', null, 1200, 256);
						this.describeAColumn(table, 'didprogori_title', 'Orientamento', null, 2200, 256);
						this.describeAColumn(table, 'didproganno_title', 'Anno di corso', null, 3100, 2048);
						this.describeAColumn(table, 'didprogporzanno_title', 'Porzione d\'anno', null, 4100, 2048);
						this.describeAColumn(table, 'aa', 'Identificativo', null, 5000, 9);
						this.describeAColumn(table, 'didproggrupp_title', 'Gruppo opzionale', null, 6300, 256);
						this.describeAColumn(table, 'insegn_codice', 'Codice Insegnamento', null, 7100, 50);
						this.describeAColumn(table, 'insegn_denominazione', 'Denominazione Insegnamento', null, 7200, 256);
						this.describeAColumn(table, 'insegninteg_codice', 'Codice Integrando', null, 8200, 50);
						this.describeAColumn(table, 'insegninteg_denominazione', 'Denominazione Integrando', null, 8300, 256);
						this.describeAColumn(table, 'attivform_start', 'Dal', null, 9000, null);
						this.describeAColumn(table, 'attivform_stop', 'Al', null, 10000, null);
						this.describeAColumn(table, 'attivform_tipovalutaz', 'Profitto o Idoneità', null, 11000, null);
						this.describeAColumn(table, 'XXcanale', 'Canali', null, 14000, null);
						this.describeAColumn(table, 'XXattivformcaratteristica', 'Caratteristiche dell\'attività formativa', null, 15000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["aa", "idsede", "iddidprog", "idattivform", "iddidprogori", "idcorsostudio", "iddidproganno", "iddidprogcurr", "iddidprogporzanno"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "didprogcurr_title asc , didprogori_title asc , didproganno_title asc , didprogporzanno_title asc , didproggrupp_title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('attivformdefaultview', new meta_attivformdefaultview('attivformdefaultview'));

	}());
